bitcoin/script/script.h

```diff
@@ -587,4 +573,27 @@ struct CScriptWitness
 
 bool CheckMinimalPush(const std::vector<unsigned char>& data, opcodetype opcode);
 
+/** Build a script by concatenating other scripts, or any argument accepted by CScript::operator<<. */
+template<typename... Ts>
+CScript BuildScript(Ts&&... inputs)
+{
+    CScript ret;
+
+    ([&] (Ts&& input) constexpr {
+        if constexpr (std::is_same_v<std::remove_reference_t<Ts>, CScript>) {
+            // If it is a CScript, extend ret with it. If ret is empty, move or copy it instead.
+            if (ret.empty()) {
+                ret = std::forward<Ts>(input);
+            } else {
+                ret.insert(ret.end(), input.begin(), input.end());
+            }
+        } else {
+            // Otherwise invoke CScript::operator<<.
+            ret << std::forward<Ts>(input);
```

Comment:
The `std::forward<Ts>` here is actually unnecessary as well, because `CScript::operator<<` never cares about the reference status of the right hand argument.